### ISSUE TITLE
Fine-tuning on custom dataset

### ISSUE THREAD

**Issue by XiaoweiLinXL** [openvla/openvla#312]
Hi community,

I am new to large model for robotics so I would love to start with the OG project.

I tried to deploy the OpenVLA model out of the box to a Trossen WidowX 250s arm for the task of picking up an toy eggplant. It worked, but the performance is not good enough, the success rate is ~10%. So I decided to fine-tune the model on custom dataset. However, after fine-tuning, the performance is really bad. Before fine-tuning, the robot at least move to the target even though it doesn't really pick it up. After fine-tuning, it doesn't even move to the target. I don't know if I missed anything during my fine-tuning pipeline, so I would like to get help from the community. I have the following questions:

1. I record the data in 10 Hz. Every timestamp, I record one image and the end-effector state=(x, y, z, r, p, y, gripper). And I calculate the action using the state of the next timestamp minus the state of the current timestamp (except for the gripper) to get action=(dx, dy, dz, dr, dp, dy, gripper). Is it the correct way to get the action? What should be the unit for each dimension? I record the data to an .npy file and the following are two steps of them.
`{'image': ..., 'state': array([2.94e-01, 1.13e-03, 1.25e-01, 2.71e-02,
       1.51e+00, 2.97e-02, 1.00e+00]), 'action': array([-2.21e-04, -4.56e-04, -1.11e-02, -4.19e-04,
        6.39e-07, -2.01e-03,  1.00e+00]), 'language_instruction': 'pick up eggplant'}, {'image': ..., 'state': array([2.94e-01, 6.74e-04, 1.14e-01, 2.67e-02,
       1.51e+00, 2.77e-02, 1.00e+00]), 'action': array([-2.53e-04, -6.96e-06, -1.57e-02, -6.83e-04,
        9.56e-07, -7.71e-04,  1.00e+00]), 'language_instruction': 'pick up eggplant'}`

2. Build the RLDS dataset: I comment out the "wrist_image" in the _info() and _parse_example() function:
https://github.com/kpertsch/rlds_dataset_builder/blob/cff9f536c53518baa744b0f31e66d53ef445d2a1/example_dataset/example_dataset_dataset_builder.py#L34C1-L35C1
https://github.com/kpertsch/rlds_dataset_builder/blob/cff9f536c53518baa744b0f31e66d53ef445d2a1/example_dataset/example_dataset_dataset_builder.py#L113
, and also the validation set because I use all of my data to train:
https://github.com/kpertsch/rlds_dataset_builder/blob/cff9f536c53518baa744b0f31e66d53ef445d2a1/example_dataset/example_dataset_dataset_builder.py#L94
Is there anything else I need to modify? And after `tfds build --overwrite`, do I still need to go through **Add Transform for Target** for uploading the dataset?

3. I fine-tuned the model for 5000 steps and the training curves seemed to converge. <img width="1842" height="359" alt="Image" src="https://github.com/user-attachments/assets/9d3cbb50-f480-417a-9a48-48479862e4f3" />
When I deployed it, the arm doesn't even move to the target even when the scene match mostly the training set. Every step is very small and the same: `action: [-5e-06, -3e-06, 5.8e-05, -0.009206, -2.8e-05, -0.002519, 0.996078]`. I already set the unnorm_key to the one in dataset_statistics.py.

This is the inference video:

![Image](https://github.com/user-attachments/assets/ed211fb1-d140-4db2-93f2-cfbd29e21289)

This is one of the training trajectory:

![Image](https://github.com/user-attachments/assets/f1deba60-65f3-461b-a6cc-3e554c54396e)

To debug, I feed the training set's images to the model, the predicted actions actually matched the ground truth action pretty much. And I run the training trajectory from the .npy file, it does pick up the object.

4. Another thing I notice is that, when I load the fine-tuned model, there are prompts and I found there is no these files in the model folder. Does it matter?
`A new version of the following files was downloaded from https://huggingface.co/openvla/openvla-7b: - configuration_prismatic.py`
`A new version of the following files was downloaded from https://huggingface.co/openvla/openvla-7b: - modeling_prismatic.py`

Thank you very much for your help!

**Comment by yw7915873-rgb** [comment]
Hi, I saw your OpenVLA WidowX custom dataset issue. I’m exploring a China-based low-cost pipeline for collecting small custom egocentric human demonstration data for VLA / robot learning experiments.

Would task-specific pilot data be useful for testing setups like yours?
